### PR TITLE
DAT-3991 Animation of sidebar in Infobox builder

### PR DESCRIPTION
**Pull request by dianafa**
# Links

* https://wikia-inc.atlassian.net/browse/DAT-3991

## Reviewers

@RafLeszczynski @bogna @vforge

### CODE REVIEW

File: front/main/app/components/infobox-builder-edit-item-section-header.js [hunk 1]
```diff
@@ -26,6 +26,14 @@ export default Ember.Component.extend(
 			}
 		}),
 
+		textInputId: Ember.computed(function () {
```
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder-edit-item-section-header.js]: please add comment why do we need this unique ids for each input element

File: front/main/app/components/infobox-builder-sidebar-header.js [hunk 2]
```diff
@@ -5,23 +5,17 @@ export default Ember.Component.extend(
...
 
 				return type ?
 					// possible message keys: add-title, add-image, add-row, add-section-header
-					i18n.t(`main.add-${this.get('item.type')}`, {
-						ns: 'infobox-builder'
-					}) :
-					i18n.t('main.sidebar-header', {
-						ns: 'infobox-builder'
-					});
+					i18n.t(`main.add-${type}`, { ns: 'infobox-builder' }) :
```
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder-sidebar-header.js]: is a valid single line object declaration when it comes to our ESLint config?

File: front/main/app/components/infobox-builder-sidebar-header.js [hunk 3]
```diff
@@ -5,23 +5,17 @@ export default Ember.Component.extend(
...
+					i18n.t('main.sidebar-header', { ns: 'infobox-builder' });
 			}
 		),
 
-		showActionButtons: Ember.computed.bool('item'),
+		showActionButtons: Ember.computed.bool('type'),
```
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder-sidebar-header.js]: why do we change `item` to `type`?

File: front/main/app/components/infobox-builder.js [hunk 4]
```diff
@@ -58,9 +58,18 @@ export default Ember.Component.extend(
...
+				id = 0,
+				type = null;
+
+			if (activeItem) {
+				name = `infobox-builder-edit-item-${activeItem.type}`;
+				id = activeItem.type + activeItem.infoboxBuilderData.id;
+				type = activeItem.type;
+			}
+
+			return {name, id, type};
```
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder.js]: please consider changing this computed property name so it corresponds with return value + add some context on why we are returning this data?

File: front/main/app/transitions.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+/**
+ * File required for liquid-fire addon to define the animations in.
+ * @returns {void}
+ */
+export default function () {
+	const liquidFireSlideToLeftContainer = '.liquid-fire-slide-to-left-container',
+		liquidFireSlideToLeftItem = '.liquid-fire-slide-to-left-item',
+		durationOut = 1250,
```
Comment (RafLeszczynski) [inline front/main/app/transitions.js]: could we cache this magic number in single const and reuse it in both places?

File: front/main/app/transitions.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+		durationIn = 1250;
+
+	this.transition(
+		this.childOf(liquidFireSlideToLeftContainer),
+		this.use('explode', {
+			pickOld: liquidFireSlideToLeftItem,
+			use: ['wait', durationOut]
```
Comment (RafLeszczynski) [inline front/main/app/transitions.js]: does this syntax (single line array) match our ESLint guidelines?

File: front/main/app/transitions.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+		}, {
+			pickNew: liquidFireSlideToLeftItem,
+			use: ['toLeft', {duration: durationIn, easing: 'easeInSine'}]
```
Comment (RafLeszczynski) [inline front/main/app/transitions.js]: does this syntax (single line array and object) match our ESLint guidelines?

**Comment by RafLeszczynski** [comment]
Please fix linter issues and tests + add tests for new logic

**Comment by RafLeszczynski** [comment]
LGTM after build passes

File: front/main/app/transitions.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/**
+ * File required for liquid-fire addon to define the animations in.
+ * @returns {void}
+ */
+export default function () {
```
Comment (vforge) [inline front/main/app/transitions.js]: As this file seems to be global - can we name this transition and do not export it as default one? (especially that file is named `transitions`)

File: front/main/app/utils/generate-guid.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+/**
+ * create unique string with prefix as a base of the string if passed
+ *
+ * @param {string} [prefix='']
+ * @returns {string} unique string
+ */
+export default function generateGuid(prefix = '') {
+	return `${prefix}${Date.now()}`;
```
Comment (vforge) [inline front/main/app/utils/generate-guid.js]: Just a thought:
Have you considered using something beside timestamp as a factor here?
```
Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
```
This snippet is also returning a random 4-byte hex string.

**Comment by vforge** [comment]
LGTM, just minor questions :)

**Pull request merged**